### PR TITLE
test and fix log softmax stability in particles_to_delta_mixtures

### PR DESCRIPTION
**Pull request by cooijmanstim**
This fixes an issue with the log softmax computation for large negative and uniform log probabilities. In isolation the issue is that these two expressions return very different values:

```
w = -1e10 * jnp.ones([11])
w_max = w.max()
print((w - w_max) - jnp.log(jnp.sum(jnp.exp(w - w_max))))
print(w - (w_max + jnp.log(jnp.sum(jnp.exp(w - w_max)))))
```

The latter returns all zeros, which is not a normalized log probability vector.

### CODE REVIEW

**Review by DanWaxman — APPROVED** [review]
This looks great, thanks Tim!
